pinot-common/src/main/java/org/apache/pinot/common/metadata/segment/SegmentZKMetadata.java

```diff
@@ -123,6 +123,14 @@ public void setIndexVersion(String indexVersion) {
     setValue(Segment.INDEX_VERSION, indexVersion);
   }
 
+  public int getZkStatVersion() {
+    return _znRecord.getVersion();
+  }
+
+  public void setZkStatVersion(int zkStatVersion) {
```

Comment:
Let's remove this and always set the version in the caller side

Reply:
+1. This setter method won't necessarily get called anywhere since this version is already set in ZNode.

Reply:
I am not sure about this point actually. I briefly looked into the Helix codebase. My understanding is that the data and stat.version are returned in different places?

Reply:
You may check `ZKHelixDataAccessor.getProperty()` within the Helix code, where the version within the `ZNRecord` is set from the `Stat` (it is reserved for that purpose).